mesonbuild/dependencies/base.py

```diff
@@ -495,31 +493,53 @@ def __init__(self, name, environment, kwargs, language=None):
         # stored in the pickled coredata and recovered.
         self.pkgbin = None
 
```

Comment:
The pkg-config code below is now the best version of binary look up I'd next abstract and use everywhere. In particular, it now supports caching in cross and native alike, checks multiple defaults like compilers, and has many detailed debug log messages.